### PR TITLE
Add testing: roslint, catkin lint, unit test

### PR DESCRIPTION
**Pull request by davetcoleman**
- Run and address roslint
 - Run and address catkin lint
 - Fix build warnings
 - Create initial test

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -151,13 +152,28 @@ install(DIRECTORY resources DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
...
+  ${PROJECT_NAME}_test
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
 
 #############
 ## Testing ##
 #############
 
-# TODO(mlautman) :-)
+#if(CATKIN_ENABLE_TESTING)
```
Comment (mlautman) [inline CMakeLists.txt]: remove or uncomment

File: config_robot/panda_grasp_data.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+base_link: 'world'
+# =====================================================
+hand:
+    end_effector_name: 'hand'  #ee group name
+
+    #actuated joints in end effector
+    joints : ['left_robotiq_2f_85_right_driver_joint', 'left_robotiq_2f_85_right_driver_joint']
```
Comment (mlautman) [inline config_robot/panda_grasp_data.yaml]: These are not the correct joints for the panda

File: src/grasp_generator.cpp [hunk 3]
```diff
@@ -126,6 +126,9 @@ bool GraspGenerator::generateCuboidAxisGrasps(const Eigen::Affine3d& cuboid_pose
       alpha_z = M_PI;
       object_width = width;
       break;
+    default:
```
Comment (mlautman) [inline src/grasp_generator.cpp]: why was this moved above Z_AXIS?

Comment (davetcoleman) [reply src/grasp_generator.cpp]: moved to top for cleaner code
